encoder/generator.go

```diff
@@ -546,17 +602,42 @@ func tsprintField(v cue.Value) (string, error) {
 		switch s := v.Source().(type) {
 		case *ast.StructLit:
 			return "", valError(v, "nested structs are not yet supported")
-		case *ast.Field:
+		case *ast.Field, nil:
```

Comment:
when rebuild a struct, apparently it is nil when we do value.Source().(type) wtf